hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/spark/SparkYarnRunnerBuilder.java

```diff
@@ -261,6 +261,41 @@ public YarnRunner getYarnRunner(String project, String sparkUser,
     builder.addToAppMasterEnvironment("SPARK_YARN_MODE", "true");
     builder.addToAppMasterEnvironment("SPARK_YARN_STAGING_DIR", stagingPath);
     builder.addToAppMasterEnvironment("SPARK_USER", jobUser);
+
+    //Set TensorFlowOnSpark required environment variables
+    if (jobType == JobType.TFSPARK) {
+
+      //Should always be false in case of TFoS
+      dynamicExecutors = false;
+
+      //No point in retrying since clusterspec is static
+      addSystemProperty(Settings.SPARK_MAX_APP_ATTEMPTS, "1");
+
+      //This is needed to solve a bug where the driver is able to allocate GPUs
+      builder.addToAppMasterEnvironment("CUDA_VISIBLE_DEVICES", "''");
+
+      //The following configuration is based on:
+      //https://github.com/yahoo/TensorFlowOnSpark/wiki/GetStarted_YARN
+
+      //IMPORTANT, if TFoS still can't find cuda libraries there may be issues with cuda installation
+      // 1. ssh to machine where the container failed
+      // 2. Make sure /usr/local/cuda exists and is a symlink pointing to e.g. /usr/local/cuda-8.0
+      // 3. Make sure /etc/ld.so.conf.d directory on the host has an entry pointing to /usr/local/cuda/lib64
+      // 4. Run 'sudo ldconfig'
+
+      String binCuda = "/usr/local/cuda/bin";
```

Comment:
We look in the "Settings.java" file for all paths and IP addresses (settings are stored in the database and built up during chef installation, but cached in a singleton bean 'Settings' to reduce load on the db).

Replace "/usr/local/cuda" with 
settings.getCudaDir()
Same goes for libCuda and libJVM below.